### DISCUSSION TITLE
Is SCS capable of finding sql injection in repository pattern?

### DISCUSSION THREAD

**Discussion by SGRedzheb** [security-code-scan/security-code-scan#268] [Q&A]
Hello.  For the example bellow SCS is not finding any vulnerability because of the use of an interface (I think so). Is there a way to configure the analyzer to propagate the taint to the concrete implementation of  the interface in any way, or is it possible at all?
 If we change the type of  `_sampleRepository` : 
 from `private ISampleRepository _sampleRepository = new SampleRepository();` 
 to     `private SampleRepository _sampleRepository = new SampleRepository();` 
 the analyzer will report a vulnerability  at ` new SampleContext().Database.ExecuteSqlCommand(input);`

```
using System.Data.Entity;
using System.Web.Mvc;

namespace SQLiSample.Controllers
{
    public class SampleController : Controller
    {
        private ISampleRepository _sampleRepository = new SampleRepository();

        public void Run()
        {
            var input = Request.RawUrl;
            _sampleRepository.ExecuteCommand(input);
        }
    }

    public interface ISampleRepository
    {
        void ExecuteCommand(string input);
    }

    public class SampleRepository : ISampleRepository
    {
        public void ExecuteCommand(string input)
        {
            new SampleContext().Database.ExecuteSqlCommand(input); // SQL injection should be reported
        }
    }

    public class SampleContext : DbContext
    {
        public DbSet<string> Test { get; set; }
    }
}
```

**Comment by dbalikhin** [comment]
I didn't work with MVC for quite a long time, but how is the  `Run` method invoked? My guess is the app cannot find a suitable source, so probably `var input = Request.RawUrl;` is not considered as tainted at all.

Update: My bad, `Run` is a valid entry point from the tool perspective. Request.RawUrl is considered tainted.
